src/NuGet.Core/NuGet.Protocol/Plugins/InboundRequestProcessingHandler.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Protocol.Plugins
+{
+    internal class InboundRequestProcessingHandler : IDisposable
+    {
+        private readonly ISet<MessageMethod> _fastProccessingMethods;
+        private readonly Lazy<DedicatedAsynchronousProcessingThread> _processingThread;
+        private bool _isDisposed;
+
+        public InboundRequestProcessingHandler() :
+            this(new HashSet<MessageMethod>())
+        {
+        }
+
+        /// <summary>
+        /// Requests from the processing methods provided in this set are handled on a dedicated thread.
+        /// </summary>
+        /// <param name="fastProcessingMethods"></param>
+        public InboundRequestProcessingHandler(ISet<MessageMethod> fastProcessingMethods) 
```

Comment:
I'm not crazy about the idea that `InboundRequestProcessingHandler` will hold onto an instance of `ISet<MessageMethod>` that is externally mutable.

Reply:
It'd require pulling in System.Collections.Immutable to get an ImmutableHashSet...unless you have other ideas?

Reply:
Change the parameter type to `IEnumerable<MessageMethod>`.  In the constructor initialize a private `HashSet<MessageMethod>` with the argument.